sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md

```diff
@@ -1,142 +1,148 @@
-|---|---|---|
-| `PartitionSender.send(...)` | `EventHubProducerAsyncClient.send()` | [Publishing events to a specific partition][PublishEventsToSpecificPartition] |
-| `EventHubClient.send(...)` | `EventHubProducerAsyncClient.send()` | [Publishing events][PublishEventsWithCustomMetadata] |
-
-### Receiving events
-
-| In v3 | Equivalent in v5 | Sample |
-|---|---|---|
-| `PartitionReceiver.receive()` | `EventHubConsumerAsyncClient.receiveFromPartition()` | [Consuming events][ConsumeEvents] |
-| `PartitionReceiver.setReceiveHandler()` | `EventHubConsumerAsyncClient.receiveFromPartition()` | [Consuming events][ConsumeEvents] |
+```java
+public static void main(String[] args) throws Exception {
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(4);
+    CompletableFuture<EventHubClient> clientFuture =
+        EventHubClient.createFromConnectionString("connection-string-for-an-event-hub",
+        scheduler);
+
+    CompletableFuture<Void> workFuture = clientFuture
+    .thenComposeAsync(client -> {
+        // Do things with the client.
+        // Dispose of client.
+        return client.close();
+    });
 
-### Minor renames
+    workFuture.get();
+    scheduler.shutdown();
+  }
+```
 
-| In v3 | Equivalent in v5 |
-|---|---|
-| `EventPosition.fromStartOfStream()` | `EventPosition.earliest()` |
-| `EventPosition.fromEndOfStream()` | `EventPosition.latest()` |
+In v5:
 
-## Migration samples
+```java
+public static void main(String[] args) throws Exception {
+    EventHubProducerAsyncClient client = new EventHubClientBuilder()
+        .connectionString("connection-string-for-an-event-hub")
+        .buildAsyncProducerClient();
 
-- [Migrating code from `PartitionSender` to `EventHubProducerAsyncClient` for sending events to a partition](#migrating-code-from-partitionsender-to-eventhubproducerasyncclient-for-sending-events-to-a-partition)
-- [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for sending events using automatic routing](#migrating-code-from-eventhubclient-to-eventhubproducerasyncclient-for-sending-events-using-automatic-routing)
-- [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for sending events with partition key](#migrating-code-from-eventhubclient-to-eventhubproducerasyncclient-for-sending-events-with-partition-key)
-- [Migrating code from `PartitionReceiver` to `EventHubConsumerAsyncClient` for receiving events in batches](#migrating-code-from-partitionreceiver-to-eventhubconsumerasyncclient-for-receiving-events-in-batches)
-- [Migrating code from `EventProcessorHost` to `EventProcessorClient` for receiving events](#migrating-code-from-eventprocessorhost-to-eventprocessorclient-for-receiving-events)
+    // Do things with the client.
+    // Dispose of client.
+    client.close();
+  }
+```
 
-### Migrating code from `PartitionSender` to `EventHubProducerAsyncClient` for sending events to a partition
+### Send events to a specific Event Hub partition
```

Comment:
Can we consider merging the two sections for specific partition and automatic routing? Most of the information is similar and we can use the same code snippet block to show both scenarios. This also gives us the opportunity to say that in both cases in v5, it is the same client and method they would use, just an additional option is required for sending to specific partition while in v2 you had to use different classes.

In this combined section, I would showcase the simple send in v5 that takes an array of events.

We can have a separate section that focuses on the batching feature.

I believe this structure would do a easier job of "selling" the benefits of v5